sdk/test-utils/perfstress/GettingStarted.md

```diff
@@ -0,0 +1,263 @@
+
+```js
+import { ServiceNameClient } from "@azure/<service-sdk>";
+import { PerfStressOptionDictionary, drainStream } from "@azure/test-utils-perfstress";
+import { `ServiceName`Test } from "./serviceNameTest.spec";
+
+export class `ServiceNameAPIName`Test extends ServiceNameTest {
+  // The next section talks about the custom options that you can provide for a test
+  public options: PerfStressOptionDictionary = {};
+
+  serviceNameClient: `ServiceName`Client;
+
+  constructor() {
+    super();
+    // Setting up the client
+  }
+
+  public async globalSetup() {
+    await super.globalSetup(); // Calling base class' setup
+    // Add any additional setup
+  }
+
+  async runAsync(): Promise<void> {
+    // call the method on `serviceNameClient` that you're interested in testing
+  }
+}
+```
+
+It is not mandatory to have separate base class and test classes. If there is nothing common among the testing scenarios of your service, feel free to merge base class with the test class to only have a single test class instead.
+
+### [Custom Options](#custom-options)
+
+As seen in the previous section, you can specify custom options along with the default options from the performance framework. You can access the options in the class using `this.parsedOptions`.
+
+Parsed options include the default options such as duration, iterations, parallel, etc offered by the perf framework as well as the custom options provided in the TestClass.
+
+```js
+interface `ServiceNameAPIName`TestOptions {
+  newOption: number;
+}
+
+export class `ServiceNameAPIName`Test extends ServiceNameTest<`ServiceNameAPIName`TestOptions> {
+  public options: PerfStressOptionDictionary<`ServiceNameAPIName`TestOptions> = {
+    newOption: {
+      required: true,
+      description: "A new option",
+      shortName: "sz",
+      longName: "newOption",
+      defaultValue: 10240
+    }
+  };
+
+  async runAsync(): Promise<void> {
+    // You can leverage the parsedOptions in the setup or globalSetup or runAsync methods as shown below.
+    // this.parsedOptions.duration.value!
+    // this.parsedOptions.newOption.value!
+  }
+}
+```
+
+## [Executing the perf tests](#executing-the-perf-tests)
+
+### [Command to run](#Command-to-run)
+
+To run a particular test, use `npm run perf-test:node` - takes the test class name as the argument along with the command line arguments you may provide.
+
+- Run `npm run perf-test:node -- TestClassName --warmup 2 --duration 7 --iterations 2 --parallel 50`
+
+### [Adding Readme/Instructions](#Adding-Readme/Instructions)
+
+Refer to [storage-blob-perf-tests-readme](https://github.com/Azure/azure-sdk-for-js/blob/fe9b1e5a50946f53b6491d7f67b2420d8ee1b229/sdk/storage/perf-tests/storage-blob/README.md) and [storage-blob-perf-tests-readme-track-1](https://github.com/Azure/azure-sdk-for-js/blob/fe9b1e5a50946f53b6491d7f67b2420d8ee1b229/sdk/storage/perf-tests/storage-blob-track-1/README.md) and have similar set of instructions for your perf project.
```

Comment:
Nothing wrong with this currently, but the `template` project would be a good spot to have a working example that's not already filled in with some other package's changes.

Reply:
.NET uses its template project for this:

https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/template/Azure.Template/perf

I recommend JS do similar.

Reply:
Good point.. logged https://github.com/Azure/azure-sdk-for-js/issues/14117

Added a good-first-issue label too. I can get to it at a later point if no one picks up.